test/apps/iodemo/io_demo.cc

```diff
@@ -1710,14 +1710,14 @@ class DemoClient : public P2pDemoCommon {
             op_info[op].num_iters++;
 
             if (is_control_iter(total_iter) && (total_iter > total_prev_iter)) {
-                /* Print performance every 1 second */
                 double curr_time = get_time();
                 if (curr_time >= (prev_time + opts().print_interval)) {
                     wait_for_responses(0);
                     if (_status != OK) {
                         break;
                     }
 
+                    curr_time = get_time();
```

Comment:
why needed?

Reply:
1. The curr_time need to be audited after waiting for all the operations are done i.e. ```wait_for_response(0)```
2. Align with the logic here: https://github.com/openucx/ucx/blob/master/test/apps/iodemo/io_demo.cc#L1734-L1736